yargs.js

```diff
@@ -1181,8 +1181,9 @@ function Yargs (processArgs, cwd, parentRequire) {
     if (!detectLocale) return
 
     try {
-      const osLocale = require('os-locale')
-      self.locale(osLocale.sync({ spawn: false }))
+      const { env } = process
+      const locale = env.LC_ALL || env.LC_MESSAGES || env.LANG || env.LANGUAGE || 'en_US'
+      self.locale(locale.replace(/[.:].*/, ''))
```

Comment:
In theory we the try / catch block is no longer needed but the was `os-locale` was used I think it was impossible for it to throw, therefore I assume that it's possible for `self.locale()` to cause an exception.